x/ecocredit/simulation/operations.go

```diff
@@ -481,6 +509,155 @@ func SimulateMsgCancel(ak ecocredit.AccountKeeper, bk ecocredit.BankKeeper,
+		if spendable == nil {
+			return op, nil, err
+		}
+
+		newAdmin, _ := simtypes.RandomAcc(r, accs)
+		if newAdmin.Address.String() == class.Admin {
+			return simtypes.NoOpMsg(ecocredit.ModuleName, TypeMsgUpdateClassAdmin, "same account"), nil, nil // skip
+		}
+
+		msg := &ecocredit.MsgUpdateClassAdmin{
+			Admin:    class.Admin,
+			ClassId:  class.ClassId,
+			NewAdmin: newAdmin.Address.String(),
+		}
+
+		txCtx := simulation.OperationInput{
+			R:               r,
+			App:             app,
+			TxGen:           simappparams.MakeTestEncodingConfig().TxConfig,
+			Cdc:             nil,
+			Msg:             msg,
+			MsgType:         msg.Type(),
+			Context:         sdkCtx,
+			SimAccount:      *account,
+			AccountKeeper:   ak,
+			Bankkeeper:      bk,
+			ModuleName:      ecocredit.ModuleName,
+			CoinsSpentInMsg: spendable,
+		}
+
+		return GenAndDeliverTxWithRandFees(txCtx)
+	}
+}
+
+// SimulateMsgUpdateClassMetadata generates a MsgUpdateClassMetadata with random metadata
+func SimulateMsgUpdateClassMetadata(ak ecocredit.AccountKeeper, bk ecocredit.BankKeeper,
+	qryClient ecocredit.QueryClient) simtypes.Operation {
+	return func(
+		r *rand.Rand, app *baseapp.BaseApp, sdkCtx sdk.Context, accs []simtypes.Account, chainID string,
+	) (simtypes.OperationMsg, []simtypes.FutureOperation, error) {
+		ctx := regentypes.Context{Context: sdkCtx}
+		class, op, err := getRandomClass(ctx, r, qryClient, TypeMsgUpdateClassMetadata)
+		if class == nil {
+			return op, nil, err
+		}
+
+		spendable, account, op, err := getAccountAndSpendableCoins(sdkCtx, bk, accs, class.Admin, TypeMsgUpdateClassMetadata)
+		if spendable == nil {
+			return op, nil, err
+		}
+
+		msg := &ecocredit.MsgUpdateClassMetadata{
+			Admin:    class.Admin,
+			ClassId:  class.ClassId,
+			Metadata: []byte(simtypes.RandStringOfLength(r, simtypes.RandIntBetween(r, 10, 256))),
+		}
+
+		txCtx := simulation.OperationInput{
+			R:               r,
+			App:             app,
+			TxGen:           simappparams.MakeTestEncodingConfig().TxConfig,
+			Cdc:             nil,
+			Msg:             msg,
+			MsgType:         msg.Type(),
+			Context:         sdkCtx,
+			SimAccount:      *account,
+			AccountKeeper:   ak,
+			Bankkeeper:      bk,
+			ModuleName:      ecocredit.ModuleName,
+			CoinsSpentInMsg: spendable,
+		}
+
+		return GenAndDeliverTxWithRandFees(txCtx)
+	}
+}
+
+// SimulateMsgUpdateClassIssuers generates a MsgUpdateClassMetaData with random values
+func SimulateMsgUpdateClassIssuers(ak ecocredit.AccountKeeper, bk ecocredit.BankKeeper,
+	qryClient ecocredit.QueryClient) simtypes.Operation {
+	return func(
+		r *rand.Rand, app *baseapp.BaseApp, sdkCtx sdk.Context, accs []simtypes.Account, chainID string,
+	) (simtypes.OperationMsg, []simtypes.FutureOperation, error) {
+		ctx := regentypes.Context{Context: sdkCtx}
+		class, op, err := getRandomClass(ctx, r, qryClient, TypeMsgUpdateClassIssuers)
+		if class == nil {
+			return op, nil, err
+		}
```

Comment:
Can't we move this part as well into the util?

Reply:
We need batch-info in other operations([SimulateMsgRetire](https://github.com/regen-network/regen-ledger/blob/master/x/ecocredit/simulation/operations.go#L332-L340)) before `getAccountAndSpendableCoins`.